### PR TITLE
DOC: prep 1.9.0 release notes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* transcribe entries from the [wiki page](https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.9.0) (the history of this page
will be used to update again before branching)
* draft in results from author/list scripts
* some initial manual curation of the author list via `.mailmap` adjustments
* manually update the notes based on merged PRs with `enhancement` label (for
those cases where it struck me as pertinent to mention)
* the "highlights" are still open to adjustment of course; simply
copied from the wiki for now--`meson` comes to mind depending on where
we land near the end of the week?

### CODE REVIEW

File: doc/release/1.9.0-notes.rst
Comment (tylerjereddy) [inline doc/release/1.9.0-notes.rst:229]: Maybe we can fuse the two "Other" themed sections?

**Comment by tylerjereddy** [comment]
CI correctly highlights a bunch of doc build warnings in the new release notes.

File: doc/release/1.9.0-notes.rst
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:34]: Maybe rephrase to read "Tensor-product spline interpolation modes were added to `scipy.interpolate.RegularGridInterpolator`" or some such.

File: doc/release/1.9.0-notes.rst
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:69]: Now that gh-16220 is in, let's add

- `BivariateSpline`s have grown a new method `partial_derivative` which constructs a new spline object representing a derivative of an original spline. This mirrors the corresponding functionality for univariate splines, `splder` and `BSpline.derivative`, and can substantially speed up repeated evaluation of derivatives.

**Review by ev-br — COMMENTED** [review]
Two small tweaks --- let me know if you prefer to have them in the wiki  instead Tyler.

File: doc/release/1.9.0-notes.rst
Comment (tupui) [inline doc/release/1.9.0-notes.rst:253]: @h-vetinari @j-bowhay FYI if there are other entries to add, please add them here as suggestion so that it's easier for Tyler.

File: doc/release/1.9.0-notes.rst
Comment (tupui) [inline doc/release/1.9.0-notes.rst:224]:
```suggestion
centering operations.

- Add `scipy.stats.qmc.PoissonDisk` to sample using the Poisson disk sampling method. It guarantees that samples are separated from each other by a given ``radius``.

Other
```

File: doc/release/1.9.0-notes.rst
Comment (h-vetinari) [reply doc/release/1.9.0-notes.rst:253]: I've tried to keep the wiki up-to-date with the issues in https://github.com/scipy/scipy/issues/15765 as they were closed. To the best of my knowledge, the list as below is complete.

Comment (tupui) [reply doc/release/1.9.0-notes.rst:253]: Thanks for confirming.

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:44]: I'd say that this doesn't belong here, it should go in the `Other changes` section, like the build system changes. `New features` is for subpackages only.

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:48]: add "(if desired for interactive use, it's not actually recommended for code, see `:ref:`scipy-api`).

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:36]: I have the Meson release note changes in the PR to change to it by default - but not a highlight. Would be good to add that here indeed.

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:80]: minor: underline length needs to match title length

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:119]: formatting: double backticks for `copy`

Comment (rgommers) [reply doc/release/1.9.0-notes.rst:119]: and same for `copy=False` below

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:189]: Just noticing this now, `fit` is a _very_ generic name. I can't remember it being proposed/discussed. Will comment on gh-15436.

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:229]: Yes, good idea.

File: doc/release/1.9.0-notes.rst
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:333]: Also FITPACK routines in `scipy.interpolate`, which power `splrep`, `splev` etc, and `*UnivariateSpline` and `*BivariateSpline` classes.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:319]: Email [here](https://mail.python.org/archives/list/scipy-dev@python.org/thread/QEMS6VZSBHTRB6OVSDBULJNNOE7PFAOV/#QEMS6VZSBHTRB6OVSDBULJNNOE7PFAOV), discussion [here](https://github.com/scipy/scipy/pull/15945). This was definitely a bug.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:615]:
```suggestion
```
This wasn't actually merged; it was superseded. Perhaps the code that generates this list could use a little tweak? Not too important, though.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:218]:
```suggestion
- faster random variate generation for ``gennorm`` and ``nakagami``
```
gh-15360

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:203]:
```suggestion
Boost for improved reliability.
```

**Comment by tylerjereddy** [comment]
I think I addressed all the comments in the PR so far, and `make html-scipyorg` passes locally (using the old `setuptools` approach). I've also included a fix to `gh_lists.py` that ignores PRs that were never actually merged (i.e., the closed PRs @mdhaber pointed out are no longer appended now). You can see the revisions in the second commit in the PR, `307443`. I also rebased to pull in the latest `meson` stuff just merged.

There will still be one or two more merges to include here and I'll check the wiki again before finalizing here/branching. I'll check the html version of the relnotes produced locally as well if I can.

**Comment by tylerjereddy** [comment]
From the html preview locally, I'd say one or two sections below the fold look incorrectly formatted (`Expired Deprecations`, `Lazy access to subpackages`; the former looks wrong, the latter my impression is that there may have been intention for the snippets not to be inline like that?), but the rest is reasonably professional looking. Perhaps less severe, but the `levy_stable` bullet point is so verbose that it almost looks incorrectly formatted, but maybe that one is "ok".

<details>

image

image



</details>

File: doc/release/1.9.0-notes.rst
Comment (WarrenWeckesser) [inline doc/release/1.9.0-notes.rst:250]: You need a blank line before the beginning of the bulleted list.  (I was going to link to the comment about this in the "missing bits" document, but apparently the link to that document in the rendered docs was broken by some recent reshuffling of the docs files. I see if I can fix that and submit a PR.)

Comment (WarrenWeckesser) [reply doc/release/1.9.0-notes.rst:250]: Here's the link to the relevant section of the docs from a PR that fixes the reference to the "missing bits" file: [Bulleted lists](https://output.circle-artifacts.com/output/job/7265df6e-f1f4-4e37-9ca6-2cd6a7b903f1/artifacts/0/html-scipyorg/dev/contributor/missing-bits.html#bulleted-lists)

**Comment by tupui** [comment]
@tylerjereddy before branching, there is an issue with the version switcher url (saying this here in case you wanted to branch before I make a PR tmr). Since the URL of the doc changed/got fixed, it also need to be adapted on the switcher. And we need to add 1.8.1.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:312]:
```suggestion
  Hence, `MultinomialQMC.random` output has now the correct shape ``(n, pvals)``.
- Several function-specific warnings (``F_onewayConstantInputWarning``,
  ``F_onewayBadInputSizesWarning``, ``PearsonRConstantInputWarning``, 
  ``PearsonRNearConstantInputWarning``, ``SpearmanRConstantInputWarning``, and
  ``BootstrapDegenerateDistributionWarning``) have been replaced with more
  general warnings.
```

**Comment by tylerjereddy** [comment]
This is the last open milestoned PR, so branching is imminent. I'm iterating on some final relnotes updates/fixes locally now.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:192]:
```suggestion
  points. Note: ``bits`` does not affect the output dtype.
```

**Comment by tylerjereddy** [comment]
@mdhaber I committed your most recent suggestion, feel free to self-commit those small ones though, I hope to merge this and branch if I see that the CI is passing when I check a bit later.

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:227]:
```suggestion
  of ``scipy.optimize.linprog`` are deprecated. Methods ``highs``, ``highs-ds``,
```
I'm not sure if this will link properly.

**Comment by tylerjereddy** [comment]
Updated the lists to account for last-minute gh-16241

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:127]:
```suggestion
- Add an ``order` `parameter to `scipy.spatial.transform.Rotation.from_quat` 
  and `scipy.spatial.transform.Rotation.as_quat` to specify quaternion format.
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:139]:
```suggestion
- Several `scipy.stats` functions support new ``axis`` (integer or tuple of
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:143]:
```suggestion
  a `scipy.stats.mstats` counterpart. Edge cases for multidimensional arrays,
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:232]:
```suggestion
- The function `spatial.distance.kulsinski` has been deprecated in favor
  of `spatial.distance.kulczynski1`.
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:251]:
```suggestion
- Passing ``radius=None`` to `scipy.spatial.SphericalVoronoi` now raises an
  error (not adding ``radius`` defaults to 1, as before)
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:256]:
```suggestion
  `scipy.signal.convolve2d` now raises an error
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:292]:
```suggestion
- The default method of `scipy.optimize.linprog` is now ``'highs'``, not
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:257]:
```suggestion
- `scipy.spatial.distance` now enforces that the input vectors are
```

File: doc/release/1.9.0-notes.rst
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:118]:
```suggestion
  * ``lo`` results in the format of the ``LinearOperator``.
```

**Review by mdhaber — APPROVED** [review]
I'm sure there are many single VS double backtick issues still present, but I should probably stop so as not to hold up release.

**Comment by tylerjereddy** [comment]
Ok, the docs are building/passing in CI so in it goes. I don't think we need to flush Azure an Nth time for release notes typo fixes, etc.

File: doc/release/1.9.0-notes.rst
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Indeed, see `tools/gh_lists.py`.

Grabbing issues based on date of closure isn't going to work, because a lot are closed because fixed in older releases, invalid reports, etc.

As of now, you should set milestones on issues that you close. A way to improve the script could be to parse commit messages for "Closes xxx", but even that isn't going to be foolproof since I often see that people fail to add that info to commit messages and only add it to PR descriptions.

Comment (ev-br) [reply doc/release/1.9.0-notes.rst:554]: A possible automation could be to auto-add a milestone to an issue which is closed by a PR  with a milestone. Maybe that's a feature request to github or maybe it's possible to do with their api.

Comment (mdhaber) [reply doc/release/1.9.0-notes.rst:554]: I started to do this for issues I was involved with. I thought I should make sure I'm doing the right thing before going further. For instance, I added the milestone to gh-14622 because it was a legitimate issue that was closed by a PR, whereas I did not add a milestone to gh-15160 because it was reported due to a misunderstanding. LMK if I should keep going this way.

Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Yes, that sounds right, thanks Matt!

Comment (mdhaber) [reply doc/release/1.9.0-notes.rst:554]: OK, done. I added a lot of milestones. 
I accidentally created a 1.90 milestone in the process. Do you know how to remove that from the list?

Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Thanks Matt. I deleted the 1.90 milestone.

**Pull request merged**
